### PR TITLE
Add sycl build target

### PR DESCRIPTION
**Pull request by Solaryee**
It is a sub PR of https://github.com/openxla/xla/pull/9042.
Then we can build xla with sycl support like: `bazel build --config=sycl xxx`

### CODE REVIEW

**Comment by ddunl** [comment]
Sorry for the delay in review here. Thanks for the PR! I imagine this is going to require a patch on our side, so this may take a while to merge.

**Review by ddunl — APPROVED** [review]

**Comment by ddunl** [comment]
As of right now I think the PR as structured is fine, it's just a build system difference internally vs externally that's the cause for the patch. Totally our responsibility to figure out. I'll ping here though if I find a change to this PR that could be helpful. I'll try to have the patch done by Friday. Thanks again!!

**Comment by kamaljeeti** [comment]
Hi @ddunl , any update on this PR? Thanks.

File: third_party/tsl/third_party/gpus/sycl_configure.bzl [hunk 1]
```diff
@@ -0,0 +1,547 @@
+"""Repository rule for SYCL autoconfiguration.
+
+`sycl_configure` depends on the following environment variables:
```
Comment (ddunl) [inline third_party/tsl/third_party/gpus/sycl_configure.bzl:3]: Could we list which environment variables are depended on? My patch is done now, so just waiting on internal approvals, sorry again for the delay!!

Comment (ShengYang1) [reply third_party/tsl/third_party/gpus/sycl_configure.bzl:3]: Great to hear the patch is done. I have updated the environment variables in comments.

**Comment by kamaljeeti** [comment]
Hi @ddunl , can you look into this once? Thanks.

**Comment by Zantares** [comment]
This is PR is merged by another PR: https://github.com/openxla/xla/pull/11271. It can be closed now.

**Pull request closed without merging**